### PR TITLE
feat: implement swap native token from erc20 contract

### PR DESCRIPTION
**Pull request by dreamer-zq**
Closed: #399

### CODE REVIEW

File: modules/token/keeper/evm_hook.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		//  SwapToNative(address from, string to, uint256 amount)
+		if len(log.Topics) != 1 {
+			continue
+		}
+
+		// Check if event is included in ERC20
+		eventID := log.Topics[0]
+		event, err := erc20.EventByID(eventID)
+		if err != nil {
+			continue
```
Comment (zhangyelong) [inline modules/token/keeper/evm_hook.go:43]: What are the err scenarios here? Do we need to print err logs and roll back transactions?

File: modules/token/keeper/evm_hook.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		}
+
+		// Check if event is a `SwapToNative` event.
+		if event.Name != contracts.EventSwapToNative {
+			hook.k.Logger(ctx).Info("emitted event", "name", event.Name, "signature", event.Sig)
```
Comment (zhangyelong) [inline modules/token/keeper/evm_hook.go:48]: Maybe most of the events are not `EventSwapToNative`, is this `info` log necessary?

File: modules/token/keeper/evm_hook.go (hunk 1 above)
Comment (dreamer-zq) [reply modules/token/keeper/evm_hook.go:43]: We only care about the events defined in our erc20 contract. We don’t have to worry about the user’s contract events, so even if there is an error in parsing, we don’t need to care.

File: modules/token/keeper/evm_hook.go
Comment (zhangyelong) [inline modules/token/keeper/evm_hook.go:1]: Additional Question: if we rollback an evm tx on the `hook` with errors, how can users get to know the errors

Comment (dreamer-zq) [reply modules/token/keeper/evm_hook.go:1]: Errors will appear in cosmos transactions

**Review by zhangyelong — APPROVED** [review]

**Review by mitch1024 — APPROVED** [review]

**Pull request merged**